pages/hold/request/[id].tsx

```diff
@@ -0,0 +1,240 @@
+import Head from "next/head"
+import {
+  Heading,
+  List,
+  Form,
+  FormField,
+  RadioGroup,
+  Radio,
+  ButtonGroup,
+  Button,
+} from "@nypl/design-system-react-components"
+
+import Layout from "../../../src/components/Layout/Layout"
+import {
+  PlainTextElement,
+  LinkedDetailElement,
+} from "../../../src/components/BibPage/BibDetail"
+import { SITE_NAME, BASE_URL, PATHS } from "../../../src/config/constants"
+
+import { findItemInBibResult } from "../../../src/utils/bibUtils"
+import { fetchBib } from "../../../src/server/api/bib"
+import initializePatronTokenAuth, {
+  doRedirectBasedOnNyplAccountRedirects,
+  getLoginRedirect,
+} from "../../../src/server/auth"
+import { getPatronData } from "../../api/account/[id]"
+
+import Bib from "../../../src/models/Bib"
+import Item from "../../../src/models/Item"
+
+import bibDetailStyles from "../../../styles/components/BibDetails.module.scss"
+
+import type { DiscoveryBibResult } from "../../../src/types/bibTypes"
+import type { DiscoveryItemResult } from "../../../src/types/itemTypes"
+
+interface HoldRequestPropsType {
+  discoveryBibResult: DiscoveryBibResult
+  discoveryItemResult: DiscoveryItemResult
+  isAuthenticated?: boolean
+}
+
+/**
+ * The Hold Request page renders a form to place a hold on an item.
+ * It renders some basic info about the bib and item based on the bib and item ids in the url,
+ * including pickup location options rendered as radio buttons in the form.
+ */
+export default function HoldRequestPage({
+  discoveryBibResult,
+  discoveryItemResult,
+  isAuthenticated,
+}: HoldRequestPropsType) {
+  const metadataTitle = `Item Request | ${SITE_NAME}`
+
+  const bib = new Bib(discoveryBibResult)
+  const item = new Item(discoveryItemResult, bib)
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    console.log("submitting form")
```

Comment:
I think you can try to mock the error and success state in this PR. So this can display the error `Banner` on the first click and then go to the confirmation route on the second click. It'll be mocked submissions but it can help push the UI and userflow along.

Reply:
And for the error state, you can pass a mocked message to the `FeedbackBox` component and open it when the "contact us" button (that looks like a link) is clicked.

Reply:
Good suggestions, the confirmation page has its own ticket but I can add the error state UI.

Reply:
The confirmation page UI is its own ticket, but for this work it's enough to go to the confirmation page, even if it's broken.